### DISCUSSION TITLE
Liquidsoap without re-encoding?

### DISCUSSION THREAD

**Discussion by mykauf** [savonet/liquidsoap#2122] [General]
Hello,
I did not like to re-encode my stream with liquidsoap. How to do it?
Thank you very much!

**Comment by toots** [comment]
Hi!

You can do that with the `%ffmpeg` encoder using audio and/or video copy:
```ruby
s = input.rtmp("rtmp://..")

output.srt(s, %ffmpeg(format="mpegts", %audio.copy, %video.copy), ...)
```
For instance.

If you have a more explicit need, we can help further. Also, I'd recommend using the latest `v2.0.2-preview` branch, which will come out soon. We have fixed bugs for that use-case in it recently.

**Comment by mykauf** [comment]
Hi toots,
thank you very much! But I have no rtmp stream, only mp3 files stored locally. Are there perhaps other options?

**Reply by smimram** [reply]
It would work with local mp3 files, simply replace the first line by something like
```ruby
s  = playlist("/my/mp3/directory")
```

**Reply by mykauf** [reply]
Okay, and how is the stream sent to an Icecast server? In the function "output.srt()" no Icecast is mentioned.

**Reply by toots** [reply]
Here's an example for `output.icecast`:
```ruby
s = playlist("/my/mp3/directory")

encoder = %ffmpeg(
  format="mp3",
  %audio.copy
)

output.icecast(
  fallible=true,
  mount="test",
  icy_metadata="true",
  format="audio/mpeg",
  encoder,
  s
)
```
Couple of notes:
* You need to pass some internal details such as `format` (mime type) and `icy_metadata`. This is because encoded formats are opaque and so auto-detection does not work. `icy_metadata` should be `"true"` unless you are using the `ogg` container.
* Likewise, you need to specify the `format` for `%ffmpeg`. This is the encapsulation layer. This requires that you make sure that all your files are of the same format, here `mp3` but it could be `aac`, `ogg/opus` etc, as long as it's consistent.

Last, I recommend to get start with the freshly released `2.0.2`, this release brings up some fixes for that kind of usage.

Hope this helps!
